File: LoopConstruction/C++/Armstrong.cpp
#include <iostream>
#include <cmath>
using namespace std;

int main() {
    int num, sum = 0, temp, remainder, n;

    cout << "Enter a number: ";
    cin >> num;

    temp = num;
    n = 0;

    while(temp != 0) {
        temp /= 10;
        ++n;
    }

    temp = num;

    while(temp != 0) {
        remainder = temp % 10;
        sum += pow(remainder, n);
        temp /= 10;
    }

    if(sum == num) {
        cout << num << " is an Armstrong number." << endl;
    } else {
        cout << num << " is not an Armstrong number." << endl;
    }

    return 0;
}

File: LoopConstruction/Java/Armstrong.java

import java.util.Scanner;

public class Armstrong {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.println("Enter a value : ");
        int n = sc.nextInt();
        int check = n;

        int count=0;

        while(n!=0){
            n/=10;
            count++;
        }

        n = check;
        int result = 0;
        while(n!=0){
           int temp  = n %10;
           result = (int) (result + Math.pow(temp, count));
            n/=10;
        }

        if(check == result)
            System.out.println("It is Armstrong No ");
        else    
            System.out.println("It is not Armstrong No ");

    }
}
